internal/controller/children.go

```diff
@@ -160,21 +174,34 @@ func (c ChildResources) Reconcile(ctx context.Context, clnt client.Client, cr *a
 		Service:     c.Service.Reconcile(ctx, clnt, cr),
 		Ingress:     c.Ingress.Reconcile(ctx, clnt, cr),
 	}
+	dataSrouceUpdates := []ChildResourceUpdate[v1.PersistentVolumeClaim]{}
+	for _, pvc := range c.DataSourcesPVCs {
+		dataSrouceUpdates = append(dataSrouceUpdates, pvc.Reconcile(ctx, clnt, cr))
+	}
+	output.DataSourcesPVCs = dataSrouceUpdates
 	return output, output.combineErrors()
 }
 
 func (c ChildResourceUpdates) AllEqual(op controllerutil.OperationResult) bool {
 	ingressOK := c.Ingress.Manifest == nil || (c.Ingress.Manifest != nil && c.Ingress.UpdateResult == op)
-	return ingressOK && c.Service.UpdateResult == op && c.PVC.UpdateResult == op && c.StatefulSet.UpdateResult == op
+	dataSourcesOK := true
+	for _, ds := range c.DataSourcesPVCs {
+		dataSourcesOK = dataSourcesOK && (ds.UpdateResult == op)
+	}
+	return ingressOK && c.Service.UpdateResult == op && c.PVC.UpdateResult == op && c.StatefulSet.UpdateResult == op && dataSourcesOK
 }
 
 func (c ChildResourceUpdates) IsRunning(pod *v1.Pod) bool {
-	onlyStatusUpdates := c.AllEqual(controllerutil.OperationResultUpdatedStatusOnly)
-	noUpdates := c.AllEqual(controllerutil.OperationResultNone)
+    // TODO: Try to re-enable the two checks below and potentially use them to determine readiness.
+	// Currently the resources created by the operator have slight changes that k8s itself applies in a few places outside
+    // of the status field. So these are picked up by the functions below. For example a PVC or a statefulset gets automatic
```

Comment:
Just to keep things neatly aligned
```suggestion
	// TODO: Try to re-enable the two checks below and potentially use them to determine readiness.
	// Currently the resources created by the operator have slight changes that k8s itself applies in a few places outside
	// of the status field. So these are picked up by the functions below. For example a PVC or a statefulset gets automatic
```